src/selectors/entities/users.test.js

```diff
@@ -458,19 +458,8 @@ describe('Selectors.Users', () => {
         // Should show full name since preferences is being used and LockTeammateNameDisplay is false
         assert.deepEqual(Selectors.makeGetDisplayName()(newTestState, testUser1.id), 'First Last');
 
-        newTestState = {
-            ...newTestState,
-            entities: {
-                ...newTestState.entities,
-                general: {
-                    ...newTestState.entities.general,
-                    config: {
-                        ...newTestState.entities.general.config,
-                        LockTeammateNameDisplay: 'true',
-                    },
-                },
-            },
-        };
+        newTestState = JSON.parse(JSON.stringify(newTestState));
```

Comment:
FWIW, I actually think the old approach was "cleaner", vs. exploiting the side effects of serialization here. Even better, perhaps, would be a separate test case that just duplicates the state definition altogether, vs. trying to "mutate".

Reply:
Great idea! I just made a `describe(..)` with many `it(..)` and just had a new state in each one. Was better than mutating the state and making comments cause now I was able to write proper descriptions in the `it(..)`. Thanks for the explanation on the mutation stuff as well. Appreciate it 😄